src/core/Akka.Tests/Actor/AskSpec.cs

```diff
@@ -66,52 +66,63 @@ protected override void OnReceive(object message)
         }
 
         [Fact]
-        public void Can_Ask_actor()
+        public async Task Can_Ask_actor()
         {
             var actor = Sys.ActorOf<SomeActor>();
-            actor.Ask<string>("answer").Result.ShouldBe("answer");
+            var res = await actor.Ask<string>("answer");
+            res.ShouldBe("answer");
         }
 
         [Fact]
-        public void Can_Ask_actor_with_timeout()
+        public async Task Can_Ask_actor_with_timeout()
         {
             var actor = Sys.ActorOf<SomeActor>();
-            actor.Ask<string>("answer",TimeSpan.FromSeconds(10)).Result.ShouldBe("answer");
+            var res = await actor.Ask<string>("answer", TimeSpan.FromSeconds(10));
+            res.ShouldBe("answer");
         }
 
         [Fact]
-        public void Can_get_timeout_when_asking_actor()
+        public async Task Can_get_timeout_when_asking_actor()
         {
             var actor = Sys.ActorOf<SomeActor>();
-            Assert.Throws<AggregateException>(() => { actor.Ask<string>("timeout", TimeSpan.FromSeconds(3)).Wait(); });
+            await Assert.ThrowsAsync<AskTimeoutException>(async () => await actor.Ask<string>("timeout", TimeSpan.FromSeconds(3)));
         }
 
         [Fact]
-        public void Can_cancel_when_asking_actor()
-        {            
+        public async Task Can_cancel_when_asking_actor()
+        {
             var actor = Sys.ActorOf<SomeActor>();
-            var cts = new CancellationTokenSource(TimeSpan.FromSeconds(3));
-            Assert.Throws<AggregateException>(() => { actor.Ask<string>("timeout", Timeout.InfiniteTimeSpan, cts.Token).Wait(); });
-            Assert.True(cts.IsCancellationRequested);
```

Comment:
No need to test CancellationToken source here - it working just fine.

Reply:
Agree